code/modules/bitrunning/components/glitch.dm

```diff
@@ -1,35 +1,36 @@
 /datum/component/glitch
 	/// Ref of the spawning forge
 	var/datum/weakref/forge_ref
+	var/datum/weakref/distort_component
 
 /datum/component/glitch/Initialize(obj/machinery/quantum_server/server, obj/machinery/byteforge/forge)
 	if(!isliving(parent))
 		return COMPONENT_INCOMPATIBLE
 
-	RegisterSignal(forge, COMSIG_MACHINERY_POWER_RESTORED, PROC_REF(on_forge_power_restored))
-	RegisterSignals(forge, list(COMSIG_MACHINERY_BROKEN, COMSIG_MACHINERY_POWER_LOST), PROC_REF(on_forge_broken))
-	forge_ref = WEAKREF(forge)
-
 	var/mob/living/owner = parent
-	server.remove_threat(owner) // so the server doesn't dust us
+	var/health_boost = 0
+
+	if (forge)
+		RegisterSignal(forge, COMSIG_MACHINERY_POWER_RESTORED, PROC_REF(on_forge_power_restored))
+		RegisterSignals(forge, list(COMSIG_MACHINERY_BROKEN, COMSIG_MACHINERY_POWER_LOST), PROC_REF(on_forge_broken))
+		forge_ref = WEAKREF(forge)
+		server.remove_threat(owner) // so the server doesn't dust us
+		health_boost = ROUND_UP(server.threat * 0.2)
 
 	owner.faction.Cut()
 	owner.faction += list(ROLE_GLITCH)
 
-	var/current_max = owner.maxHealth + ROUND_UP(server.threat * 0.2)
-	owner.maxHealth = clamp(current_max, 200, 500)
+	owner.maxHealth = clamp(owner.maxHealth + health_boost, 200, 500)
 	owner.fully_heal()
+	owner.create_digital_aura()
 
-	var/atom/thing = owner
-	thing.create_digital_aura()
-
-/// Sakujo
-/datum/component/glitch/proc/dust_mob()
-	if(QDELETED(parent))
-		return
+/datum/component/glitch/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_LIVING_DEATH, PROC_REF(on_death))
+	distort_component = WEAKREF(parent.AddComponent(/datum/component/holographic_nature))
 
-	var/mob/living/owner = parent
-	owner.dust()
+/datum/component/glitch/UnregisterFromParent()
+	UnregisterSignal(parent, COMSIG_LIVING_DEATH)
+	QDEL_NULL(distort_component)
```

Comment:
```suggestion
	qdel(distort_component.resolve())
	distort_component = null
```

Reply:
This is handled by `/datum/weakref/Destroy`
as far as I know deleting a weakref to also delete its reference if it exists (and otherwise just delete the weakref) should be perfectly fine